Docs: Update HTTP API docs for nested folders (#67412)
grafana/grafana: docs/sources/developers/http_api/folder.md

Returns all folders that the authenticated user has permission to view. You can control the maximum number of folders returned through the `limit` query parameter, the default is 1000. You can also pass the `page` query parameter for fetching folders from a page other than the first one.

> If nested folders are enabled, the operation expects an additional optional query parameter `parentUid` with the parent folder UID and returns the immediate subfolders that the authenticated user has permission to view.

Reviewer: ```suggestion
If nested folders are enabled, the operation expects an additional optional query parameter `parentUid` with the parent folder UID, and returns the immediate subfolders that the authenticated user has permission to view.
```

> If the parameter is not supplied then it returns immediate subfolders under the root

Reviewer: In the first and second sentences, what is "it" replacing? The call? The operation? We should spell that out.

Author: I have re-phrased it: can you check once again? thank you!

> If the parameter is not supplied then the operation returns immediate subfolders under the root

Reviewer: ```suggestion
If the parameter is not supplied, then the operation returns immediate subfolders under the root
```

}
```

> If nested folders are enabled, and the folder is nested (lives under another folder) then the response additionally contains:

Reviewer: Same thing for lines 196 and 267. 

```suggestion
If nested folders are enabled, and the folder is nested (lives under another folder), then the response additionally contains:
```

> - **parentUid** - the parent folder UID.
> - **parents** - an array with the whole tree hierarchy starting from the root going down up to the parent folder.

Reviewer: Should "down" and "up" both be in the description of **parents**? Or are we just missing a comma?
Whatever fix we make here should be applied to lines 198, 199, 269, and 270.

```suggestion
- **parentUid** - The parent folder UID.
- **parents** - An array with the whole tree hierarchy, starting from the root going down up to the parent folder.
```

Will return the folder identified by id.

> This is deprecated. Please use [get folder by UID]({{< ref "#get-folder-by-uid" >}}) instead.

Reviewer: ```suggestion
This is deprecated. Use [get folder by UID]({{< ref "#get-folder-by-uid" >}}) instead.
```

- **401** – Unauthorized
- **403** – Access Denied
- **404** – Folder not found

## Move folder

`POST /api/folders/:uid/move`

> Will move the folder.

Reviewer: ```suggestion
Moves the folder.
```

This is relevant only if nested folders are enabled.

**Required permissions**

See note in the [introduction]({{< ref "#folder-api" >}}) for an explanation.

If moving the folder under another folder:

| Action          | Scope                                  |
| --------------- | -------------------------------------- |
| `folders:write` | `folders:uid:<destination folder UID>` |

If moving the folder under root:
| Action | Scope |
| -------------- | ------------- |
| `folders:create` | `folders:*` |

> JSON Body schema:

Reviewer: ```suggestion
JSON body schema:
```

> - **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If it's empty then the folder is moved under the root.

Reviewer: ```suggestion
- **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If this is empty, then the folder is moved under the root.
```

> - **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If it's empty then the folder is moved under the root.

Reviewer: The link on line 425 produces a 404.

> - **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If this is empty, then the folder is moved under the root.

Reviewer: ```suggestion
- **parentUid** – Optional [unique identifier]({{< relref "#identifier-id-vs-unique-identifier-uid" >}}) of the new parent folder. If this is empty, then the folder is moved under the root.
```

Accept: application/json
Content-Type: application/json
Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk

{
  "parentUid": "d80b18c0-266a-4aa4-ad5d-5537a00cb8e8",
}
```

**Example Response**:

```http
HTTP/1.1 200
Content-Type: application/json

{
	"id": 4,
	"uid": "a5393ec3-5568-4e88-8809-b866968ae8a6",
	"title": "just-testing",
	"url": "/dashboards/f/a5393ec3-5568-4e88-8809-b866968ae8a6/just-testing",
	"hasAcl": false,
	"canSave": true,
	"canEdit": true,
	"canAdmin": true,
	"canDelete": true,
	"createdBy": "Anonymous",
	"created": "2023-04-27T21:55:01.593741+03:00",
	"updatedBy": "Anonymous",
	"updated": "2023-04-27T21:55:15.747444+03:00",
	"parentUid": "d80b18c0-266a-4aa4-ad5d-5537a00cb8e8",
	"parents": [
		{
			"id": 2,
			"uid": "d80b18c0-266a-4aa4-ad5d-5537a00cb8e8",
			"title": "f0",
			"url": "",
			"hasAcl": false,
			"canSave": true,
			"canEdit": true,
			"canAdmin": true,
			"canDelete": true,
			"createdBy": "Anonymous",
			"created": "2023-04-27T21:53:46.070672+03:00",
			"updatedBy": "Anonymous",
			"updated": "2023-04-27T21:53:46.070673+03:00"
		}
	]
}
```

Status Codes:

- **200** – Moved

> - **400** – Errors (invalid json, missing or invalid fields, etc)

Reviewer: ```suggestion
- **400** – Errors (invalid JSON, missing or invalid fields, and so on)
```

- **401** – Unauthorized

> - **403** – Access Denied
> - **404** – Not Found

Reviewer: ```suggestion
- **403** – Access denied
- **404** – Not found
```
